platform/darwin/src/MGLStyle.mm

```diff
@@ -584,6 +586,19 @@ - (MGLTransition)transition
     return transition;
 }
 
+- (void)setLight:(MGLLight *)light
+{
+    std::unique_ptr<mbgl::style::Light> mbglLight = std::make_unique<mbgl::style::Light>([light toMBGLLight]);
+    self.mapView.mbglMap->setLight(std::move(mbglLight));
+}
+
+- (MGLLight *)light
+{
+    mbgl::style::Light mbglLight = *self.mapView.mbglMap->getLight();
```

Comment:
This code appears to obtain a double pointer to the mbgl::style::Light and assign it to a Light, in all likelihood creating a copy. Would it be possible to pass the original pointer returned by getLight() directly into -initWithMBGLLight:?